cmd/apps/cockroachdb_app.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) arkade author(s) 2021. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallCockroachdb() *cobra.Command {
+	var cockroachdb = &cobra.Command{
+		Use:          "cockroachdb",
+		Short:        "Install cockroachdb",
+		Long:         "Install cockroachdb",
+		Example:      ` arkade app install cockroachdb`,
+		SilenceUsage: true,
+	}
+	cockroachdb.Flags().String("namespace", "default", "Namespace for the app")
+	cockroachdb.Flags().Bool("persistence", false, "Use a 100Gi Persistent Volume to store data")
+
+	cockroachdb.Flags().Bool("single-node", false, "Run CockroachDB instances in standalone mode with replication disabled, so the StatefulSet does NOT FORM A CLUSTER")
+	cockroachdb.Flags().Int64("replicas", 1, "Satetfulset replica count")
+
+	cockroachdb.Flags().Bool("enable-tls", false, "Whether to run securely using TLS certificates")
+
+	cockroachdb.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set tls.enabled=false)")
+
+	cockroachdb.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
+			return err
+		}
+
+		namespace, _ := command.Flags().GetString("namespace")
+		persistence, _ := command.Flags().GetBool("persistence")
+		singleNode, _ := command.Flags().GetBool("single-node")
+		enableTls, _ := command.Flags().GetBool("enable-tls")
+		replicas, _ := command.Flags().GetInt64("replicas")
+
+		overrides := map[string]string{}
+
+		if singleNode {
+			overrides["conf.single-node"] = "true"
+		}
+		if enableTls {
+			overrides["tls.enabled"] = "false"
+		}
+		if persistence {
+			overrides["storage.persistentVolume.enabled"] = "true"
+		}
+		overrides["statefulset.replicas"] = fmt.Sprintf("%d", replicas)
+
+		customFlags, _ := command.Flags().GetStringArray("set")
+
+		if err := config.MergeFlags(overrides, customFlags); err != nil {
+			return err
+		}
+
+		cockroachdbOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmRepo("cockroachdb/cockroachdb").
+			WithHelmURL("https://charts.cockroachdb.com/").
+			WithOverrides(overrides).
+			WithKubeconfigPath(kubeConfigPath)
+
+		_, err := apps.MakeInstallChart(cockroachdbOptions)
+		if err != nil {
+			return err
+		}
+
+		println(cockroachdbInstallMsg)
+		return nil
+	}
+
+	return cockroachdb
+}
+
+const CockroachdbInfoMsg = `# Get started at: https://www.cockroachlabs.com/docs/stable
+
+# If you used tls check that the secrets were created on the cluster
+
+kubectl get secrets
```

Comment:
What is the name of the secret, is it stable and can we add it here?

Reply:
Those are the self signed tls certs, that cockroach gives you if you use tls auth for accessing the database itself.

Reply:
Same question: what is the name of the secret, can you add the specific name here please, `kubectl get secrets` will not necessarily make it obvious.

Look at how we do this in the openfaas app for instance.

Reply:
Sorry I missunderstood the question, sure I will provide a more detailed command here, thanks for the notice! :)

Reply:
`kubectl get secrets` doesn't help the user do the right thing, it'd almost be better to omit it.

If we can get the specific secret name and decode it for them it'd be more intuitive and a better experience for a first-time user.